docs/spec/spec.md

```diff
@@ -1,219 +1,166 @@
-    }
-
-    function close() returns error? {
-        return self.persistClient.close();
-    }
-}
-
-public class MedicalNeedStream {
-
-    private stream<anydata, error?> anydataStream;
-
-    public isolated function init(stream<anydata, error?> anydataStream) {
-        self.anydataStream = anydataStream;
-    }
-
-    public isolated function next() returns record {|entities:MedicalNeed value;|}|error? {
-        var streamValue = self.anydataStream.next();
-        if streamValue is () {
-            return streamValue;
-        } else if (streamValue is error) {
-            return streamValue;
-        } else {
-            record {|entities:MedicalNeed value;|} nextRecord = {value: check streamValue.value.cloneWithType(entities:MedicalNeed)};
-            return nextRecord;
-        }
-    }
-
-    public isolated function close() returns error? {
-        return self.anydataStream.close();
-    }
-}
-```
+## 3. Generating Persistence Derived Types and Clients
 
-In addition, a `database_configuration.bal` file will be generated in the same directory with the configurable variables related to database configurations as follows.
-```ballerina
-configurable int port = ?;
-configurable string host = ?;
-configurable string user = ?;
-configurable string database = ?;
-configurable string password = ?;
+```bash
+bal persist generate
 ```
 
-## 4. Creating Database Tables
-Users can define database entities in their Ballerina projects. They can create database tables corresponding to these entities by executing `bal persist push` command. Users can then use the generated client objects to perform operations on these tables programmatically without having to write SQL statements.
-
-Consider the following entity record inside the Ballerina project.
-```ballerina
-import ballerina/time;
-import ballerina/persist;
-
-@persist:Entity {
-    key: ["needId"],
-    tableName: "MedicalNeeds"
-}
-public type MedicalNeed record {|
-    @persist:AutoIncrement
-    readonly int needId = -1;
-
-    int itemId;
-    int beneficiaryId;
-    time:Civil period;
-    string urgency;
-    int quantity;
-|};
+The command will generate [Derived Entity Types and Persist Clients](https://github.com/ballerina-platform/module-ballerina-persist/blob/main/docs/spec/spec.md#3-derived-entity-types-and-persist-clients)
+as per the `persist` specification.
+
+It will add generated files under the conventions,
+1. If the file name is the same as the package name, it will generate the files under the `default` module.
+   ```
+   medical-center
+   ├── generated
+         ├── database_configuration.bal
+         ├── generated_client.bal
+         └── generated_types.bal
+   ├── persist
+         └── medical-center.bal
+   ├── Ballerina.toml
+   ├── Config.toml
+   └── main.bal
+   ```
+2. If the file name is different to the package name, it will generate the files under a new submodule with the same name as the file.
```

Comment:
```suggestion
2. If the file name is different from the package name, it will generate the files under a new submodule with the same name as the file.
```